### PR TITLE
feat: integrate SonarCloud, clean directory, and start overview screen

### PR DESCRIPTION
**Pull request by veridisyndrome**

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Gradle packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build and analyze
```
Comment (veridisyndrome) [inline .github/workflows/CI.yaml:33]: - name: Set Gradlew permissions
        run: chmod +x gradlew

**Review by veridisyndrome — COMMENTED** [review]
Maybe we should add this?

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (jeongJ2004) [reply .github/workflows/CI.yaml:33]: There is "All checks have failed" and when we click on details, there is this error message : 
Run ./gradlew build sonar --info
/home/runner/work/_temp/c3644348-b754-42e3-ba8e-021b2b[9](https://github.com/BrewR-EPFL/BrewR/actions/runs/11212842619/job/31164749183?pr=26#step:6:10)865ee.sh: line 1: ./gradlew: Permission denied
Error: Process completed with exit code 126.

Comment (jeongJ2004) [reply .github/workflows/CI.yaml:33]: So when you run/build those codes, it built in your computer ? 
I'm afraid to approve this PR and merging the one with errors.

Comment (veridisyndrome) [reply .github/workflows/CI.yaml:35]: https://sonarcloud.io/summary/new_code?id=BrewR-EPFL_BrewR&pullRequest=26

File: app/src/main/java/com/android/brewr/SimpleData.kt
Comment (tsogtb) [inline app/src/main/java/com/android/brewr/SimpleData.kt:1]: SecondActivity was deleted but not deleted in the AndroidManifest.xml

Comment (veridisyndrome) [reply app/src/main/java/com/android/brewr/SimpleData.kt:1]: Sorry, you can commit the change!

Comment (veridisyndrome) [reply app/src/main/java/com/android/brewr/SimpleData.kt:1]: Please use Coventional Commits as I asked you in the Slack:)

**Comment by tsogtb** [comment]
I think we also need to cache the gradle folder to make the CI run faster

**Comment by veridisyndrome** [comment]
After approval, do not delete the branch, I am still going to use it.

**Comment by tsogtb** [comment]
I added the OverviewTest and edited the CI file so that SonarCloud now has the coverage. We don't have enough code to have meaningful tests at the moment and the Quality gate is not passing because of the condition coverage. I don't seem to have the permissions to change the quality gate. If you could change the quality gate we can merge it and @jeongJ2004 can create his PR I think.

**Review by tsogtb — APPROVED** [review]

**Comment by tsogtb** [comment]
> I added the OverviewTest and edited the CI file so that SonarCloud now has the coverage. We don't have enough code to have meaningful tests at the moment and the Quality gate is not passing because of the condition coverage. I don't seem to have the permissions to change the quality gate. If you could change the quality gate we can merge it and @jeongJ2004 can create his PR I think.

Oh I didnt know we could bypass the sonarcloud requirements

**Comment by veridisyndrome** [comment]
> > I added the OverviewTest and edited the CI file so that SonarCloud now has the coverage. We don't have enough code to have meaningful tests at the moment and the Quality gate is not passing because of the condition coverage. I don't seem to have the permissions to change the quality gate. If you could change the quality gate we can merge it and @jeongJ2004 can create his PR I think.
> 
> Oh I didnt know we could bypass the sonarcloud requirements

@tsogtb I already told you multiple times to use conventional commits. So please do it. Even more when you commit on my branch.

**Comment by tsogtb** [comment]
To be fair I had to push so many times to trigger the CI and fix SonarCloud so it was easier to comment "ci". Other than that I think that the previous commit messages more or less fall into the category "if applied this commit will.." but thanks for pointing it out

**Pull request merged**